src/client/grid-view.js

```diff
@@ -1,539 +1,408 @@
+     * @param playerToPlay quel joueur va jouer
+     * @param frame quelle manche de jeu (de 1 à NB_FRAMES maximum)
+     * @param nbThrow n° de lancer 1, 2 ou 3 si la dernière manche
+     */
+    generatePossibilities() {//TODO générer en fonction des coups précédents
+        let currentFrame = this.gridview['player'][this.playerNameToPlay]['frames'][this.frameToPlay-1];
+        let nbKeelDown = 0;
+        if (this.throwToPlay == 2 && currentFrame['c1'] != this.NB_KEELS)
+            nbKeelDown = currentFrame['c1'] != null ? currentFrame['c1'] : 0;
+
+        for (let i = 0; i <= this.NB_KEELS - nbKeelDown; i++) {
+            let button = document.createElement("button");
+            if (this.frameToPlay != this.NB_FRAMES) {
+                if (this.throwToPlay == 1 && i == this.NB_KEELS)
+                    button.innerHTML = "X";
+                else if (this.throwToPlay == 2 && i == this.NB_KEELS - nbKeelDown)// si on a fait un strike au coup 1, on ne rentrera pas dedans car le backed changera le coup à jouer
+                    button.innerHTML = "/";
+                else
+                    button.innerHTML = i.toString();
+            } else {
+                // console.log("last frame nbthrow " + this.throwToPlay );
+                if (this.throwToPlay == 1 && i == this.NB_KEELS)
+                    button.innerHTML = "X";
+                else if (this.throwToPlay == 2) {
+                    // console.log(this.NB_KEELS - currentFrame['c1'])
+                    // console.log(i)
+                    //dernier coup pas un strike et le nb de quilles tombées est égal au nb de quilles restantes
+                    if (currentFrame['c1'] != this.NB_KEELS && i == this.NB_KEELS - currentFrame['c1'])
+                        button.innerHTML = "/";
+                    else if (currentFrame['c1'] == this.NB_KEELS && i == this.NB_KEELS)
+                        button.innerHTML = "X";
+                    else
+                        button.innerHTML = i.toString();
+                }
+                else if (this.throwToPlay == 3 && i == this.NB_KEELS)
+                    button.innerHTML = "X";
+                else
+                    button.innerHTML = i.toString();
+            }
 
-}
-//TODO TEMPORARY FIX
-function total() {
-    for (let i = 1; i < rowNumber; i++) {
-        for (let j = 1; j <= 10; j++) {
-            let left = document.getElementById(getId(i, j, CellType.LEFT)).textContent;
-            let right = document.getElementById(getId(i, j, CellType.RIGHT)).textContent;
-            let bottom = document.getElementById(getId(i, j, CellType.BOTTOM)).textContent;
-            let center = document.getElementById(getId(i, j, CellType.CENTER)).textContent;
-            grid["Player"+i](i, j, left, right, bottom, center);
+            button.addEventListener("click", function() {
+                console.log("Click on :", i);
+                //request aussi
```

Comment:
Why haven't you add the request to the related API ?

Reply:
couldn't test at the moment